providers/fcontext.rb

```diff
@@ -0,0 +1,50 @@
+# Support whyrun
+def whyrun_supported?
+  true
+end
+
+use_inline_resources
+
+# Create if doesn't exist, do not touch if fcontext is already registered
+action :add do
+  escaped_file_spec = Regexp.escape(new_resource.file_spec)
+  noregex_file_spec = new_resource.file_spec.split(/(\(\/\.\*\)\?|\.\*)$/)[0]
```

Comment:
Your idea is fine, but it seems a lot of fuss (also - making convergence decisions based on files present during compilation seems fishy).

I'd really like to use find with regex (try `find -regex '/usr/lib(64)?/vmware/lib(/.*)?/HConfig.so'`). Problems are:

1. `find` searches the entire FS first (I get "access denied" on files that don't even resemble the path) - should be mitigated by extracting the safest directory
2. There are several regex types (as indicated by `-regextype` in the manpage). I couldn't find the type `restorecon` uses, and I'm wary of discovering later that we've used the wrong one.

I admit issue 1 is the solution we're trying to find anyway, though :)
I think our issue is extracting a path from the regex, doesn't have to be the narrowest because we'll additionally use the regex thing. We can give up on it completely too, I guess.
Opinions?